### DISCUSSION TITLE
Home Assistant Addon for valetudo-mapper and MQTT Questions

### DISCUSSION THREAD

**Discussion by ottelo9** [rand256/valetudo#404] [General]
Hi there. Is there a home assistant addon for the Valetudo-RE "valetudo-mapper" ? Or how could I add it to my hassio home assistant? Then I want to control the roborock s5 via mqtt. MQTT is running und my robo is listed inside the MQTT Integration. Is there a tutorial how to send the robo via mqtt for room/zone cleaning?

**Comment by pidator** [comment]
I can't help you concerning the HA addons since I don't have a HA instance running but the MQTT Api and the available commands are documented [here](https://github.com/rand256/valetudo/wiki/MQTT-API) .

**Comment by ottelo9** [comment]
No one is using HA?

**Comment by ottelo9** [comment]
I found this one: https://github.com/Poeschl/Hassio-Addons/tree/master/ICantBelieveItsNotValetudo

**Reply by pidator** [reply]
This will not work with Valetudo RE because it's for Valetudo Vanilla. The source of this addon is the corresponding Vanilla service [ICantBelieveItsNotValetudo](https://github.com/Hypfer/ICantBelieveItsNotValetudo). For RE you'll have to use [valetudo-mapper](https://github.com/rand256/valetudo-mapper) to generate a png of the map published by mqtt.

Do you tried the  corresponding addon of @Poeschl for valetudo-mapper:
https://github.com/Poeschl/Hassio-Addons/tree/master/valetudo-mapper ? As far as I understand, it should provide the same service like your linked addon, but this time for a running RE instance...

**Reply by ottelo9** [reply]
Big thanks! The valetudo-mapper works!
